commands/utility/nickall.js

```diff
@@ -0,0 +1,73 @@
+const { SlashCommandBuilder, PermissionFlagsBits } = require("discord.js");
+/*  ----------------------
+Google Sheets Setup
+-------------------------- */
+const { spreadsheetId } = require("../../config.json");
+const { google } = require("googleapis");
+
+const auth = new google.auth.GoogleAuth({
+	keyFile: "./credentials.json",
+	scopes: "https://www.googleapis.com/auth/spreadsheets"
+})
+const sheetClient = auth.getClient();
+const googleSheets = google.sheets({ version: "v4", auth: sheetClient });
+/* ------------------- */
+
+module.exports = {
+	data: new SlashCommandBuilder()
+		.setName("nickall")
+    	.setDescription("Sets the nickname of all users according to the Electrium spreadsheet data"),
+    	// .setDefaultMemberPermissions(PermissionFlagsBits.Administrator),
+  	async execute(interaction) {
+		const discordmembers = new Map();
+		const sheetmembers = new Map();
```

Comment:
The first map fetches the discord server members. The second one tracks the through the Google sheets. Poor and hacky design but gets the job done.

Reply:
This is only because we can't perform any interactions with a Discord username alone. We **must** use the snowflake, and a more work and time is needed to find out how to do that.